.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,34 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, 3.x ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master, 3.x ]
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+
+    - run: |
+       sudo apt-get install build-essential scons pkg-config libx11-dev libxcursor-dev libxinerama-dev libgl1-mesa-dev libglu-dev libasound2-dev libpulse-dev libxi-dev libxrandr-dev yasm
+       scons platform=x11 target=release_debug udev=no -j 4
```

Comment:
Each command should have different steps:
- Dependencies
- Compilation
- Run

etc. just like in others CI files